### PR TITLE
build: Try harder to determine FMT_VERSION

### PR DESCRIPTION
**Pull request by hhoffstaette**
fmt-11.0 moved `FMT_VERSION` from `core.h` to `base.h`, so try the new header first and then fall back to the old one. Tested against fmt-9.1, 10.2.1, 11.0.

Closes: #1477

### CODE REVIEW

File: cmake/FindFmt.cmake [hunk 1]
```diff
@@ -3,11 +3,19 @@ mark_as_advanced(FMT_INCLUDE_DIR FMT_LIBRARY)
...
-    file(READ "${FMT_INCLUDE_DIR}/fmt/core.h" _fmt_core_h)
-    string(REGEX MATCH "#define FMT_VERSION ([0-9]+)" _ "${_fmt_core_h}")
+    file(READ "${FMT_INCLUDE_DIR}/fmt/base.h" _fmt_base_h)
+    string(REGEX MATCH "#define FMT_VERSION ([0-9]+)" _ "${_fmt_base_h}")
+    if("${CMAKE_MATCH_0}" STREQUAL "")
+      file(READ "${FMT_INCLUDE_DIR}/fmt/core.h" _fmt_core_h)
+      string(REGEX MATCH "#define FMT_VERSION ([0-9]+)" _ "${_fmt_core_h}")
+    endif()
+    if("${CMAKE_MATCH_0}" STREQUAL "")
+      message(STATUS "FMT_VERSION not found")
```
[the remark this answers is not in the document]
Comment (hhoffstaette) [reply cmake/FindFmt.cmake:16]: Fixed, thanks!

[the remark this answers is not in the document]
Comment (hhoffstaette) [reply cmake/FindFmt.cmake]: Now actually fixed.. :roll_eyes:

**Comment by jrosdahl** [comment]
Thanks.

File: cmake/FindFmt.cmake [hunk 2]
```diff
@@ -3,11 +3,19 @@ mark_as_advanced(FMT_INCLUDE_DIR FMT_LIBRARY)
 if(DEP_FMT STREQUAL "BUNDLED")
   message(STATUS "Using bundled Fmt as requested")
 else()
-  find_path(FMT_INCLUDE_DIR fmt/core.h)
+  find_path(FMT_INCLUDE_DIR fmt/base.h fmt/core.h)
```
Comment (jrosdahl) [inline cmake/FindFmt.cmake:6]: Oops, there should be a `NAMES` here as well. Also, when `fmt/core.h` is found (and `fmt/base.h` is not) then `file(READ "${FMT_INCLUDE_DIR}/fmt/base.h" _fmt_base_h)` on line 9 will fail.

Fixed in 3b09afc5f792f0bd0a15cf6b8408ea40eb069787

Comment (hhoffstaette) [reply cmake/FindFmt.cmake:6]: I had a feeling that I missed something but somehow all combinations I tried worked for me during testing. Thanks!

**Pull request merged**